### PR TITLE
Update k8s dependencies

### PR DESCRIPTION
**Pull request by Neo2308**
* Updated k8s dependencies to v0.29.3
* Updated operator-framework/operator-registry to v1.39.0
* Updated kubebuilder to v3.14.2
* Updated k8s version in Makefile

Relates to: https://github.com/operator-framework/operator-sdk/issues/6651

### CODE REVIEW

**Comment by Neo2308** [comment]
@rashmigottipati  could you help review this PR?

File: go.mod [hunk 1]
```diff
@@ -6,27 +6,27 @@ require (
...
 	github.com/maxbrunsfeld/counterfeiter/v6 v6.8.1
-	github.com/onsi/ginkgo/v2 v2.14.0
-	github.com/onsi/gomega v1.30.0
-	github.com/operator-framework/operator-lib v0.12.0
-	github.com/operator-framework/operator-registry v1.35.0
-	github.com/prometheus/client_golang v1.18.0
-	github.com/prometheus/client_model v0.5.0
+	github.com/onsi/ginkgo/v2 v2.17.1
+	github.com/onsi/gomega v1.32.0
+	github.com/operator-framework/operator-lib v0.12.1-0.20240430173419-b394e34ce702
```
Comment (Neo2308) [inline go.mod:11]: @everettraven is it okay use a specific commit here? Or should we cut a new version of operator-lib?

Comment (everettraven) [reply go.mod:11]: Should be okay to use a specific commit for now, but we do need to cut a release for operator-lib

**Review by everettraven — CHANGES_REQUESTED** [review]
These changes look good, but I think we also need to make sure we update the python packages provided in the base image here: https://github.com/operator-framework/ansible-operator-plugins/blob/4e8c78d21b6017ab0a280e3d274d97edf50b8fc4/images/ansible-operator/Pipfile#L11

File: pkg/plugins/ansible/v1/scaffolds/internal/templates/config/rbac/role.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ func (f *ManagerRole) SetTemplateDefaults() error {
 	}
 
 	f.TemplateBody = fmt.Sprintf(roleTemplate, machinery.NewMarkerFor(f.Path, rulesMarker))
-
+	f.IfExistsAction = machinery.OverwriteFile
```
Comment (Neo2308) [inline pkg/plugins/ansible/v1/scaffolds/internal/templates/config/rbac/role.go:42]: Added this line due to: https://github.com/kubernetes-sigs/kubebuilder/pull/3627

File: internal/ansible/handler/logging_enqueue_owner_test.go [hunk 3]
```diff
@@ -51,7 +51,7 @@ var _ = Describe("Eventhandler", func() {
 
 		httpClient, err := rest.HTTPClientFor(cfg)
 		Expect(err).ShouldNot(HaveOccurred())
-		mapper, err = apiutil.NewDiscoveryRESTMapper(cfg, httpClient)
+		mapper, err = apiutil.NewDynamicRESTMapper(cfg, httpClient)
```
Comment (Neo2308) [inline internal/ansible/handler/logging_enqueue_owner_test.go:54]: NewDiscoveryRESTMapper was removed upstream. Ref: https://github.com/kubernetes-sigs/controller-runtime/pull/2611

File: pkg/plugins/util/cleanup.go [hunk 4]
```diff
@@ -76,12 +76,12 @@ func UpdateKustomizationsInit() error {
 		`
 # [WEBHOOK] To enable webhook, uncomment all the sections with [WEBHOOK] prefix including the one in
 # crd/kustomization.yaml
-#- manager_webhook_patch.yaml
+#- path: manager_webhook_patch.yaml
```
Comment (Neo2308) [inline pkg/plugins/util/cleanup.go:79]: patchesStrategicMerge was changed to patches upstream, causing these changes: https://github.com/kubernetes-sigs/kubebuilder/pull/3661

File: hack/generate/samples/ansible/memcached.go [hunk 5]
```diff
@@ -82,7 +82,7 @@ func addingAnsibleTask(dir string, gvk schema.GroupVersionKind) {
 // Makefile to the version specified
 func replaceKustomizeVersion(dir string, version string) {
 	err := kbutil.ReplaceInFile(filepath.Join(dir, "Makefile"),
-		"curl -sSLo - https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/v5.1.1/kustomize_v5.1.1_$(OS)_$(ARCH).tar.gz | \\",
+		"curl -sSLo - https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/v5.3.0/kustomize_v5.3.0_$(OS)_$(ARCH).tar.gz | \\",
```
Comment (Neo2308) [inline hack/generate/samples/ansible/memcached.go:85]: I am unsure if this function is needed anymore, for now updated kustomise version to v5.3.0.

File: testdata/memcached-molecule-operator/config/default/manager_auth_proxy_patch.yaml
Comment (Neo2308) [inline testdata/memcached-molecule-operator/config/default/manager_auth_proxy_patch.yaml:1]: All changed under testdata were autogenerated when running `make test-sanity`

Labels removal: https://github.com/kubernetes-sigs/kubebuilder/issues/3783
kube-rbac-proxy version update: https://github.com/kubernetes-sigs/kubebuilder/pull/3665

**Comment by Neo2308** [comment]
@everettraven could you review this PR since its blocking https://github.com/operator-framework/operator-sdk/pull/6736?

**Comment by Neo2308** [comment]
@everettraven / @rashmigottipati can you help review this PR? This blocking work on the 1.29 bump

File: go.mod [hunk 6]
```diff
@@ -6,27 +6,27 @@ require (
...
+	k8s.io/api v0.29.3
+	k8s.io/apiextensions-apiserver v0.29.3
+	k8s.io/apimachinery v0.29.3
+	k8s.io/client-go v0.29.3
+	k8s.io/kubectl v0.29.3
 	k8s.io/utils v0.0.0-20240102154912-e7106e64919e
-	sigs.k8s.io/controller-runtime v0.16.3
-	sigs.k8s.io/kubebuilder/v3 v3.12.0
+	sigs.k8s.io/controller-runtime v0.17.2
+	sigs.k8s.io/kubebuilder/v3 v3.14.1
```
Comment (acornett21) [inline go.mod:29]: Can this move to `3.14.2`, there is a bug in previous versions affecting scaffolding.

**Review by everettraven — COMMENTED** [review]
Changes LGTM aside from the comment that @acornett21 left

File: go.mod (hunk 1 above)
Comment (acornett21) [reply go.mod:11]: This could be `v0.13.0` since that is the version that has the 1.29 bump.

**Comment by Neo2308** [comment]
Made the specified changes. @everettraven, @acornett21 could you review again?

**Comment by Neo2308** [comment]
The e2e-molecule test failure seems to be due to an environment issue, I don't seem to have the permissions to retrigger it.
@joelanford could you help retrigger the checks?

**Comment by Neo2308** [comment]
@acornett21 could you help fix the build failure?

**Comment by acornett21** [comment]
@Neo2308 It looks like this is a python dep issue, I took this code as a base and raised #80 with a few changes and all seems well. I reached out on k8s slack to try to get some time to talk about this PR and #80 and #79.

**Comment by Neo2308** [comment]
Rebased the PR onto latest master. @acornett21 could you approve the workflow runs?

**Comment by Neo2308** [comment]
The sanity tests are passing locally now, @acornett21 / @everettraven could you run approve the workflow runs?

**Comment by acornett21** [comment]
@Neo2308 It doesn't appear that you generated the `lock` file correctly, you need to follow the instructions in the images/Readme file. But even if that is generated correctly there are still other issues, since version of `requests` needed  to get past the CVE does not support the use case (protocol) we are using.

https://github.com/Azure/azure-iot-sdk-python/issues/1182

Controller Logs
```
{"level":"error","ts":"2024-06-25T14:43:26Z","logger":"runner","msg":"Traceback (most recent call last):\n  File \"/usr/local/lib/python3.9/site-packages/requests/adapters.py\", line 633, in send\n    conn = self.get_connection_with_tls_context(\n  File \"/usr/local/lib/python3.9/site-packages/requests/adapters.py\", line 489, in get_connection_with_tls_context\n    conn = self.poolmanager.connection_from_host(\n  File \"/usr/local/lib/python3.9/site-packages/urllib3/poolmanager.py\", line 246, in connection_from_host\n    return self.connection_from_context(request_context)\n  File \"/usr/local/lib/python3.9/site-packages/urllib3/poolmanager.py\", line 258, in connection_from_context\n    raise URLSchemeUnknown(scheme)\nurllib3.exceptions.URLSchemeUnknown: Not supported URL scheme http+unix\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.9/site-packages/ansible_runner/__main__.py\", line 874, in main\n    res = run(**run_options)\n  File \"/usr/local/lib/python3.9/site-packages/ansible_runner/interface.py\", line 210, in run\n    r.run()\n  File \"/usr/local/lib/python3.9/site-packages/ansible_runner/runner.py\", line 118, in run\n    self.status_callback('starting')\n  File \"/usr/local/lib/python3.9/site-packages/ansible_runner/runner.py\", line 106, in status_callback\n    ansible_runner.plugins[plugin].status_handler(self.config, status_data)\n  File \"/usr/local/lib/python3.9/site-packages/ansible_runner_http/events.py\", line 35, in status_handler\n    status = send_request(plugin_config['runner_url'],\n  File \"/usr/local/lib/python3.9/site-packages/ansible_runner_http/events.py\", line 18, in send_request\n    return session.post(url_actual, headers=headers, json=(data))\n  File \"/usr/local/lib/python3.9/site-packages/requests/sessions.py\", line 637, in post\n    return self.request(\"POST\", url, data=data, json=json, **kwargs)\n  File \"/usr/local/lib/python3.9/site-packages/requests/sessions.py\", line 589, in request\n    resp = self.send(prep, **send_kwargs)\n  File \"/usr/local/lib/python3.9/site-packages/requests/sessions.py\", line 703, in send\n    r = adapter.send(request, **kwargs)\n  File \"/usr/local/lib/python3.9/site-packages/requests/adapters.py\", line 637, in send\n    raise InvalidURL(e, request=request)\nrequests.exceptions.InvalidURL: Not supported URL scheme http+unix\n","job":"7062013831693068113","name":"bootstrap-token-abcdef","namespace":"kube-system","error":"exit status 1","stacktrace":"github.com/operator-framework/ansible-operator-plugins/internal/ansible/runner.(*runner).Run.func1\n\tansible-operator-plugins/internal/ansible/runner/runner.go:269"}
```

I'm not really sure how we work around this, since I'm not well versed in python.

**Comment by acornett21** [comment]
@Neo2308 After the discussion in slack, @everettraven agrees the path forward for now is to revert the `ansible-core` changes and ignore the CVE in both docker files. I believe the ID would be `71064`.

**Comment by Neo2308** [comment]
@acornett21 made the changes

**Review by acornett21 — APPROVED** [review]

**Comment by Neo2308** [comment]
@everettraven could you review/approve this PR as well?

**Review by everettraven — APPROVED** [review]

**Pull request merged**